base/regex.jl

```diff
@@ -449,3 +449,27 @@ function hash(r::Regex, h::UInt)
     h = hash(r.compile_options, h)
     h = hash(r.match_options, h)
 end
+
+## escaping ##
+"""
+    regex_escape(s::AbstractString)
+
+Sanitize a string to make it safe for use in regular expression pattern construction. Any
+regular expression metacharacters are escaped along with whitespace.
+
+# Examples
+```jldoctest
+julia> regex_escape("Bang!")
+"Bang\\!"
+
+julia> regex_escape("  ( [ { . ? *")
+"\\ \\ \\(\\ \\[\\ \\{\\ \\.\\ \\?\\ \\*"
+
+julia> regex_escape("/^[a-z0-9_-]{3,16}\$/")
+"/\\^\\[a\\-z0\\-9_\\-\\]\\{3,16\\}\\\$/"
+```
+"""
+function regex_escape(s::AbstractString)
+    res = replace(s, r"([()[\]{}?*+\-|^\$\\.&~#\s=!<>|:])" => s"\\\1")
```

Comment:
why `=!<>|:` (which aren't in the python list)?

Reply:
I believe [my original](https://github.com/JuliaLang/julia/issues/6124#issuecomment-408627070) was based on PHP, but adding some precautionary characters from Python, and escaping whitespace.

Reply:
(But I haven’t looked at `wrap_string`. Documenting that might be a reasonable alternative. 🤷‍♂️)

Reply:
(My comments end with [this version](https://github.com/JuliaLang/julia/issues/6124#issuecomment-408630949), BTW.)

Reply:
Seems like `wrap_string` uses the Perl strategy (with `\Q`)? I just thought that was overkill, but given that it’s already used, that might be the way to go, dropping this version. (Just my two cents.)